_posts/2022-05-23-deploy-agent-on-fedora-coreos.markdown

```diff
@@ -0,0 +1,314 @@
+date: 2022-05-23 12:00:00 +1000
+categories: flotta 
+author: Cosmin Tupangiu
+tags:
+  - guide
+  - flotta
+  - fcos
+  - flotta-agent 
+
+summary-1: Deploy flotta-agent on Fedora CoreOS 35
+---
+
+During development, the user may need to spin off a large number of Edge Devices. Fedora CoreOS provides a easy way to create
+a virtual machine to deploy Edge Device on it. Moreover, the relatively low footprint of CoreOS VM allows you to have a large number of 
+Edge Devices on the same host.
+
+## General `flotta-agent` installation
+
+`Flotta-agent` can be installed on any machine using the `rpm` provided by the [fedora copr](https://copr.fedorainfracloud.org/coprs/project-flotta/).
+To install `flotta-agent`:
+```bash
+sudo dnf copr enable project-flotta/flotta 
+sudo dnf install -y node_exporter yggdrasil flotta-agent
+```
+
+Now, that `flotta-agent` has been installed on the OS, we need to configure `yggdrasil` to know how to connect to `Flotta Edge API`.
+By default, the configuration file can be found under `/etc/yggdrasil/config.toml`.
+An example of `yggdrasil` configuration file is provided below:
+```yaml
+log-level = "debug"
+cert-file = "/etc/pki/consumer/cert.pem"
+key-file = "/etc/pki/consumer/key.pem"
+ca-root = ["/etc/pki/consumer/ca.pem"]
+path-prefix = "api/flotta-management/v1"
+protocol = "http"
+server = "flotta.io:8043"
+```
+
+Let's explain line by line the configuration:
+
+- `log-level`: sets the log level for `yggdrasil`. The options are: `debug`, `info`, `warn`, `error`.
+- `cert-file`, `key-file`: path to client certification and private key.
+- `ca-root`: CA root certification file.
+- `path-prefix`: this is set to `api/flotta-management/v1`. All the request from the Edge Device will start with prefix followed by the id of the device (e.g. `/api/flotta-management/v1/data/<device-id>/in`).
+- `protocol`: set to `http`. 
+- `server`: address of the `Flotta Edge API`. The domain name has to be the same as the domain set in the cluster because the certificates are valid only for that domain.
+
+The certificates can be obtain by running `make get-certs` in `Flotta Operator` cluster.
+
+There is one line missing here, which is `client-id`. We will add later on this line cause we need the `machine-id` and this is available only after the first boot.
+
+## Creating of the ignition file
+
+CoreOS configure the system using a [ignition](https://docs.fedoraproject.org/en-US/fedora-coreos/producing-ign/) file. The `ignition` file is produced from a `butane` file.
+You can find the specification of the `butane` file on [coreos/butane](https://coreos.github.io/butane/config-fcos-v1_3/). Basically, the `butane` file is a `yaml` file which specify
+how the system will be configured after startup. 
+
+All the system configuration is done via `systemd` so to install and setup `flotta-agent` we need to create two services, one for install and one for setup. These services needs to run only one time when we are provisioning the system.
```

Comment:
> one for install and one for setup

Not sure:
one for installing and one for setting up
one to install and one to setup

WDYT?

In addition, I would suggest to reword a bit the sentence. After reading, it comes to my mind: to install and to setup what?